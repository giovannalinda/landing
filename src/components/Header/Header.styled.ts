import styled from 'styled-components'
import { theme } from '~/styles'

export const Header = styled.header`
  max-width: ${theme.grid.lg};
  width: 100%;
  margin: 0 auto;
  padding: ${theme.space[6]} 0;
`

export const Navigation = styled.nav`
  display: flex;
  align-items: center;
  justify-content: end;
  width: 100%;

  ul {
    display: flex;
  }

  li {
    font-weight: ${theme.font.weigths.bold};
    letter-spacing: ${theme.font.letterSpacing.md};
    font-size: 1.4rem;
    z-index: 1;

    a {
      color: ${theme.colors.white};
    }

    & + li {
      margin-left: ${theme.space[7]};
    }
  }
`
