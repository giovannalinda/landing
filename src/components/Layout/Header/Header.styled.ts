import styled from 'styled-components'
import { theme } from '~/styles'

export const Header = styled.header`
  max-width: ${theme.grid.lg};
  width: 100%;
  margin: 0 auto;
  padding: ${theme.space[6]} 0;
  display: flex;
  align-items: flex-start;

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    align-items: center;
    padding: ${theme.space[6]} ${theme.space[4]};
    background: ${theme.colors.secondaryPurple};
    border-radius: 0 0 ${theme.borderRadius.lg} ${theme.borderRadius.lg};
  }
`

export const Navigation = styled.nav`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  width: 100%;

  ul {
    display: flex;
  }

  li {
    font-weight: ${theme.font.weigths.bold};
    letter-spacing: ${theme.font.letterSpacing.md};
    font-size: ${theme.font.sizes.small};
    z-index: 1;

    a {
      position: relative;
      color: ${theme.colors.white};

      &::after {
        content: '';
        position: absolute;
        background: ${theme.colors.white};
        width: 0;
        height: 0.1rem;
        bottom: -${theme.space[0]};
        left: 0;
        transition: all ${theme.transitions.fast};
      }

      &:hover {
        &::after {
          width: 100%;
        }
      }
    }

    & + li {
      margin-left: ${theme.space[7]};
    }
  }

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    justify-content: center;

    li + li {
      margin-left: ${theme.space[3]};
    }
  }
`
