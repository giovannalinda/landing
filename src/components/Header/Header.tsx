import { Navigation } from './Navigation'

import * as S from './Header.styled'

export function Header() {
  return (
    <S.Header>
      <Navigation />
    </S.Header>
  )
}
