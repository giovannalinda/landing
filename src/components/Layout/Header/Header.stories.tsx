import { Meta, Story } from '@storybook/react'

import { Header } from './Header'

export default {
  title: 'Components/Layout/Header',
  component: Header,
} as Meta

export const Default: Story = () => (
  <div style={{ maxWidth: '100rem', margin: '0 auto' }}>
    <Header />
  </div>
)
