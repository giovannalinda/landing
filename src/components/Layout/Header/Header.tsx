import Link from 'next/link'
import * as S from './Header.styled'

export function Header() {
  return (
    <S.Header>
      <S.Navigation>
        <ul>
          <li>
            <Link href='#about' scroll>
              ABOUT
            </Link>
          </li>
          <li>
            <Link href='#services' scroll>
              SERVICES
            </Link>
          </li>
          <li>
            <Link href='#works' scroll>
              WORKS
            </Link>
          </li>
          <li>
            <Link href='#contact' scroll>
              CONTACT
            </Link>
          </li>
        </ul>
      </S.Navigation>
    </S.Header>
  )
}
