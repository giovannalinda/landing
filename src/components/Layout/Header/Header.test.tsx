import { render } from '@testing-library/react'
import { Header } from './Header'

describe('Header', () => {
  it('should match snapshot', () => {
    const { container } = render(<Header />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
