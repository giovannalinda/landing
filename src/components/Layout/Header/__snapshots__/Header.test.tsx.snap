// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  class="sc-bczRLJ bscaEW"
>
  <nav
    class="sc-gsnTZi kDkCl"
  >
    <ul>
      <li>
        <a
          href="/#about"
        >
          ABOUT
        </a>
      </li>
      <li>
        <a
          href="/#services"
        >
          SERVICES
        </a>
      </li>
      <li>
        <a
          href="/#works"
        >
          WORKS
        </a>
      </li>
      <li>
        <a
          href="/#contact"
        >
          CONTACT
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
